lib/pages/classes/view/class_view.dart

```diff
@@ -45,9 +49,35 @@ class _ClassViewState extends State<ClassView> {
   @override
   Widget build(BuildContext context) {
     final classProvider = Provider.of<ClassProvider>(context);
+    final feedbackProvider =
+        Provider.of<FeedbackProvider>(context, listen: false);
+    final authProvider = Provider.of<AuthProvider>(context, listen: false);
+    feedbackProvider
+        .userSubmittedFeedbackForClass(authProvider.uid, widget.classHeader.id)
+        .then((value) => alreadyCompletedFeedback = value);
 
     return AppScaffold(
       title: Text(S.current.navigationClassInfo),
+      actions: [
+        if (Utils.feedbackEnabled)
+          AppScaffoldAction(
+              icon: Icons.rate_review_outlined,
+              tooltip: S.current.navigationClassFeedback,
+              onPressed: () {
+                if (!alreadyCompletedFeedback) {
```

Comment:
Won't this break if you press the button too quickly after opening the page? Since the initial value is null, and you're populating it in an async call that may take a bit... One easy (though not ideal) solution would be to disable the button if we're still fetching data (`onPressed: alreadyCompletedFeedback == null ? null : () {...}`. and do setState where you're setting the alreadyCompletedFeedback value above.